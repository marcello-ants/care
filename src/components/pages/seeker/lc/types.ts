import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
