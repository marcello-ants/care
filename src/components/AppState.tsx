import { useRouter } from 'next/router';
import { createContext, useReducer, useContext, ReactNode, useEffect } from 'react';
import { debounce } from 'lodash-es';
import { configureScope, captureException, Severity } from '@sentry/nextjs';
import {
  initialAppState,
  rootReducer,
  isStateVersionCompatible,
  doesStateHaveDefaultKeys,
  validateState,
} from '@/state';
import patchAppState from '@/state/patch';
import { initialState as flowInitialState } from '@/state/flow';
import logger from '@/lib/clientLogger';
import { ClientLogger } from '@care/utils';
import getVertical from '@/utilities/verticalUtils';
import { SessionStorageHelper } from '@/utilities/SessionStorageHelper';
import { trackPageViewInSiftScience } from '@/utilities/siftScienceHelper';
import { AppAction, AppDispatch, AppState } from '@/types/app';
import { StateCheckingResponse } from '@/types/common';
import {
  LOCAL_STORAGE_STATE_KEY,
  CZEN_BASE_PATH,
  FLOWS,
  POST_A_JOB_ROUTES,
  CLIENT_FEATURE_FLAGS,
} from '@/constants';
import { useFeatureFlags } from '@/components/FeatureFlagsContext';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';

type AppStateProviderProps = {
  children: ReactNode;
  initialStateOverride?: AppState;
  czenJSessionId?: string | undefined;
  referrerCookie?: string | undefined | null;
  userHasAccount?: boolean;
  memberUuid?: string;
};

function getStateFromStorageOrDefault(defaultState: AppState) {
  let stateFromStorageJson;
  const sessionStorageAvailable = SessionStorageHelper.isAvailable();

  if (sessionStorageAvailable) {
    stateFromStorageJson = SessionStorageHelper.getItem(LOCAL_STORAGE_STATE_KEY);
  } else {
    logger.warn({ event: 'sessionStorageUnavailable' });
  }

  if (stateFromStorageJson) {
    const stateFromStorage = JSON.parse(stateFromStorageJson) as AppState;
    try {
      const patchedState = patchAppState(stateFromStorage, defaultState);

      if (isStateVersionCompatible(patchedState) && doesStateHaveDefaultKeys(patchedState)) {
        return patchedState;
      }
      logger.warn({ event: 'invalidStateFromLocalStorage', state: stateFromStorageJson });
    } catch (e) {
      captureException(e, { level: Severity.Warning });
    }
  }
  return defaultState;
}

function persistState(state: AppState) {
  SessionStorageHelper.setItem(LOCAL_STORAGE_STATE_KEY, JSON.stringify(state));
}

const debouncePersistState = debounce(persistState, 500, {
  leading: true,
  trailing: true,
  maxWait: 1000,
});

function reducerWrapper(state: AppState, action: AppAction) {
  const nextState = rootReducer(state, action);
  debouncePersistState(nextState);
  return nextState;
}

export function redirectIfInvalidState(state: AppState, pathname: string): StateCheckingResponse {
  if (
    typeof Intl === 'undefined' ||
    typeof Intl.NumberFormat === 'undefined' ||
    typeof Intl.DateTimeFormat === 'undefined'
  ) {
    // TODO: Remove this with SC-214
    logger.error({ event: 'IntlUnavailable', source: pathname, invalidState: state });
    return {
      redirectCalled: true,
      readableMessage: 'Safari 9 and lower is not currently supported',
    };
  }

  const isStateValid = validateState(state, pathname);
  const redirect = !isStateValid;

  if (redirect) {
    SessionStorageHelper.removeItem(LOCAL_STORAGE_STATE_KEY);

    window.location.assign(CZEN_BASE_PATH);
  }

  return { redirectCalled: redirect };
}

export function getFlowName(pathname: string, oldFlowName?: string): string {
  let newFlowName = Object.values(FLOWS).find((flow) => pathname.includes(flow.path))?.name;

  if (!newFlowName) {
    // TODO: this shouldn't be needed after SC-123 is implemented
    const isPostAJob = Object.values(POST_A_JOB_ROUTES).some((route) => pathname.includes(route));
    if (isPostAJob) {
      newFlowName = 'SEEKER_POST_A_JOB';
    }
  }

  // Don't allow shifting from in-facility to seeker flow name
  if (newFlowName === FLOWS.SEEKER.name && oldFlowName === FLOWS.SEEKER_IN_FACILITY.name) {
    return FLOWS.SEEKER_IN_FACILITY.name;
  }

  return newFlowName ?? '';
}

function useFlowDetection(dispatch: AppDispatch, oldFlowName: string) {
  const { pathname } = useRouter();
  const newFlowName = getFlowName(pathname, oldFlowName);
  const vertical = getVertical();
  ClientLogger.setLoggerStatics({ flow: newFlowName, vertical });

  useEffect(() => {
    dispatch({ type: 'setFlowName', flowName: newFlowName });

    configureScope((scope) => {
      scope.setTags({
        flow: newFlowName,
        vertical,
      });
    });
  }, [newFlowName]);
}

const AppStateContext = createContext<AppState>(initialAppState);
const AppDispatchContext = createContext<AppDispatch>(() =>
  // eslint-disable-next-line no-console
  console.warn(
    'Calling default dispatch is a no-op! You are probably consuming app dispatch in a component above AppStateProvider'
  )
);

export const AppStateProvider = ({
  children,
  initialStateOverride,
  czenJSessionId,
  referrerCookie,
  userHasAccount,
  memberUuid,
}: AppStateProviderProps) => {
  const hydratedInitialAppState = {
    ...initialAppState,
    flow: {
      ...flowInitialState,
      czenJSessionId,
      referrerCookie,
      userHasAccount,
      memberUuid,
    },
  };
  const featureFlags = useFeatureFlags();
  const sessionStorageRetrievalSuccessEvaluation =
    featureFlags?.flags[CLIENT_FEATURE_FLAGS.SESSION_STORAGE_RETRIEVAL_SUCCESS];
  const shouldMeasureSessionStorageRetrievalSuccess =
    sessionStorageRetrievalSuccessEvaluation?.variationIndex === 1;

  const router = useRouter();

  const logSessionStorageError = () => {
    logger.info({ event: 'sessionStorageStateUnavailable', path: router.pathname });
    AnalyticsHelper.logEvent({
      name: 'sessionStorageStateUnavailable',
    });
  };
  const launchDarklyWrappedGetStateFromStorageOrDefault = (defaultState: AppState) => {
    if (shouldMeasureSessionStorageRetrievalSuccess) {
      try {
        const sessionStorageState = SessionStorageHelper.getItem(LOCAL_STORAGE_STATE_KEY);
        if (sessionStorageState && sessionStorageState.length > 0) {
          const stateFromStorage = JSON.parse(sessionStorageState);
          if (stateFromStorage) {
            logger.info({ event: 'sessionStorageStateAvailable', path: router.pathname });
          } else {
            logSessionStorageError();
          }
        } else {
          logSessionStorageError();
        }
      } catch (_e) {
        logSessionStorageError();
      }
    }
    return getStateFromStorageOrDefault(defaultState);
  };

  const [state, dispatch] = useReducer(
    reducerWrapper,
    initialStateOverride ?? hydratedInitialAppState,
    launchDarklyWrappedGetStateFromStorageOrDefault
  );
  useFlowDetection(dispatch, state.flow.flowName);
  const routePath = router.asPath?.split('?')[0];
  const stateCheckResult = redirectIfInvalidState(state, routePath);
  if (stateCheckResult.redirectCalled) {
    if (stateCheckResult.readableMessage) {
      return (
        <>
          <p>Redirecting to Member Home Page...</p>
          <p>{stateCheckResult.readableMessage}</p>
        </>
      );
    }
    return <></>;
  }

  useEffect(() => {
    trackPageViewInSiftScience(state);
  }, [routePath]);

  return (
    <AppDispatchContext.Provider value={dispatch}>
      <AppStateContext.Provider value={state}>{children}</AppStateContext.Provider>
    </AppDispatchContext.Provider>
  );
};

AppStateProvider.defaultProps = {
  initialStateOverride: undefined,
  czenJSessionId: undefined,
  referrerCookie: undefined,
  userHasAccount: false,
  memberUuid: undefined,
};

export const useAppState = () => useContext(AppStateContext);
export const useAppDispatch = () => useContext(AppDispatchContext);

export const useSeekerState = () => useContext(AppStateContext).seeker;
export const useProviderState = () => useContext(AppStateContext).provider;
export const useProviderCCState = () => useContext(AppStateContext).providerCC;
export const useFlowState = () => useContext(AppStateContext).flow;
export const useSeekerCCState = () => useContext(AppStateContext).seekerCC;
export const useSeekerHKState = () => useContext(AppStateContext).seekerHK;
export const useSeekerPCState = () => useContext(AppStateContext).seekerPC;
export const useSeekerTUState = () => useContext(AppStateContext).seekerTU;
export const useLtcgState = () => useContext(AppStateContext).ltcg;
