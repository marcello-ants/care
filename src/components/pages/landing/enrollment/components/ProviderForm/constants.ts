export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
