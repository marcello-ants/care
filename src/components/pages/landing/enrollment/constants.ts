export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
