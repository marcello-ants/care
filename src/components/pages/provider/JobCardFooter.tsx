import React from 'react';
import { Button } from '@material-ui/core';
import { jobsNearAnalytics } from '@/components/pages/provider/jobPostUtil';
import FixElementToBottom from '@/components/FixElementToBottom';
import useEnterKey from '@/components/hooks/useEnterKey';

interface JobCardFooterProps {
  numberOfJobsNear: number;
  nextUrl: string;
}
export default function JobCardFooter({ numberOfJobsNear, nextUrl }: JobCardFooterProps) {
  const handleNext = () => {
    jobsNearAnalytics(numberOfJobsNear);
    window.location.assign(nextUrl);
  };
  useEnterKey(true, handleNext);
  return (
    <FixElementToBottom>
      <Button color="primary" variant="contained" size="large" fullWidth onClick={handleNext}>
        Next
      </Button>
    </FixElementToBottom>
  );
}
