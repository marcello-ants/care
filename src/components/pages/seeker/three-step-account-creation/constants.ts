import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
