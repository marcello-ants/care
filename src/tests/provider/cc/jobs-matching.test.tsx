import React from 'react';
import { render, waitFor, screen, fireEvent } from '@testing-library/react';
import { useRouter } from 'next/router';
import { ThemeProvider } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';
import { GET_NEW_JOBS_PROVIDER } from '@/components/request/GQL';
import { MockedProvider } from '@apollo/client/testing';
import JobsMatching from '@/pages/provider/cc/jobs-matching';
import JobCardFooter from '@/components/pages/provider/JobCardFooter';
import { AppStateProvider } from '@/components/AppState';
import { initialAppState } from '@/state';
import { AppState } from '@/types/app';
import { PROVIDER_CHILD_CARE_ROUTES, CZEN_BACKGROUND_CHECK_CC } from '@/constants';

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

const initialState: AppState = {
  ...initialAppState,
  providerCC: {
    ...initialAppState.providerCC,
    numberOfJobsNear: 3,
  },
};

const moreThanOneJobMatchingMock = {
  request: {
    query: GET_NEW_JOBS_PROVIDER,
    variables: {
      numResults: 3,
    },
  },
  result: {
    data: {
      getNewJobsForProvider: [
        {
          title: 'Companion Care Needed For My Mother In Round Rock',
          description:
            'We need companion care to take care of my mother in Round Rock. We would prefer someone who can handle light housekeeping. Additional responsibilities include meal preparation. We might ask you to run errands now and then. ',
          state: 'TX',
          city: 'Round Rock',
          payRange: {
            hourlyRateFrom: { amount: '14', currencyCode: 'USD', __typename: 'Money' },
            hourlyRateTo: { amount: '22', currencyCode: 'USD', __typename: 'Money' },
            __typename: 'PayRange',
          },
          jobPostDate: '2021-04-12T20:12:00.000Z',
          __typename: 'JobSummary',
        },
        {
          title: 'Hands-on Care Needed For Myself In Round Rock',
          description:
            'We need hands-on care to take care of myself in Round Rock. Personal care required, including feeding. Part of your responsibilities will include running errands. We ask that you handle transportation as needed.',
          state: 'TX',
          city: 'Round Rock',
          payRange: {
            hourlyRateFrom: { amount: '14.50', currencyCode: 'USD', __typename: 'Money' },
            hourlyRateTo: { amount: '22', currencyCode: 'USD', __typename: 'Money' },
            __typename: 'PayRange',
          },
          jobPostDate: '2021-04-15T13:12:00.000Z',
          __typename: 'JobSummary',
        },
        {
          title: 'Hands-on Care Needed For My Mother In Austin',
          description:
            'We need hands-on care to take care of myself in Round Rock. Personal care required, including feeding. Part of your responsibilities will include running errands. We ask that you handle transportation as needed.',
          state: 'TX',
          city: 'Austin',
          payRange: {
            hourlyRateFrom: { amount: '14', currencyCode: 'USD', __typename: 'Money' },
            hourlyRateTo: { amount: '22', currencyCode: 'USD', __typename: 'Money' },
            __typename: 'PayRange',
          },
          jobPostDate: '2021-04-20T16:12:00.000Z',
          __typename: 'JobSummary',
        },
      ],
    },
  },
};
const oneMatchingJobMock = {
  request: {
    query: GET_NEW_JOBS_PROVIDER,
    variables: {
      numResults: 1,
    },
  },
  result: {
    data: {
      getNewJobsForProvider: [
        {
          title: 'Hands-on Care Needed For My Mother In Austin',
          description:
            'We need hands-on care to take care of myself in Round Rock. Personal care required, including feeding. Part of your responsibilities will include running errands. We ask that you handle transportation as needed.',
          state: 'TX',
          city: 'Austin',
          payRange: {
            hourlyRateFrom: { amount: '14', currencyCode: 'USD', __typename: 'Money' },
            hourlyRateTo: { amount: '22', currencyCode: 'USD', __typename: 'Money' },
            __typename: 'PayRange',
          },
          jobPostDate: '2021-04-20T13:12:00.000Z',
          __typename: 'JobSummary',
        },
      ],
    },
  },
};
const noJobsMatchingMock = {
  request: {
    query: GET_NEW_JOBS_PROVIDER,
    variables: {
      numResults: 0,
    },
  },
  result: {},
};
const jobsNearMe = 3;
const nojobMocksList = [noJobsMatchingMock];
async function renderNoJobsRedirect(mocks: any, nextUrl: any) {
  const view = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider {...{ theme }}>
        <AppStateProvider initialStateOverride={initialState}>
          <JobCardFooter numberOfJobsNear={jobsNearMe} nextUrl={nextUrl} />
        </AppStateProvider>
      </ThemeProvider>
    </MockedProvider>,
    {}
  );
  return view;
}
const mocksList = [moreThanOneJobMatchingMock, oneMatchingJobMock];
async function renderResultAndWaitFinished(mocks: any) {
  const view = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider {...{ theme }}>
        <AppStateProvider initialStateOverride={initialState}>
          <JobsMatching />
        </AppStateProvider>
      </ThemeProvider>
    </MockedProvider>,
    {}
  );
  await waitFor(() => {
    expect(
      screen.getByText('Companion Care Needed For My Mother In Round Rock')
    ).toBeInTheDocument();
  });

  return view;
}

describe('Jobs matching page', () => {
  let mockRouter: any = null;
  const { location: originalLocation } = window;
  beforeAll(() => {
    jest.useFakeTimers('modern').setSystemTime(new Date('2021-04-20T16:00:00.000Z'));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  beforeEach(async () => {
    mockRouter = {
      push: jest.fn(),
      pathname: PROVIDER_CHILD_CARE_ROUTES.JOBS_MATCHING,
      asPath: PROVIDER_CHILD_CARE_ROUTES.JOBS_MATCHING,
    };
    (useRouter as jest.Mock).mockReturnValue(mockRouter);
    // @ts-ignore
    delete window.location;
    /* eslint-disable no-restricted-globals */
    (window.location as Pick<typeof window.location, 'assign'>) = {
      assign: jest.fn(),
    };
  });
  afterEach(() => {
    mockRouter = null;
    window.location = originalLocation;
  });
  it('matches snapshot', () => {
    const view = render(
      <ThemeProvider theme={theme}>
        <JobCardFooter numberOfJobsNear={jobsNearMe} nextUrl={CZEN_BACKGROUND_CHECK_CC} />
      </ThemeProvider>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });
  it('query should redirect when there are 0 jobs', async () => {
    await renderNoJobsRedirect(nojobMocksList, CZEN_BACKGROUND_CHECK_CC);
    const nextBtn = screen.getByRole('button', { name: 'Next' });
    fireEvent.click(nextBtn);
    await waitFor(() => {
      expect(window.location.assign).toHaveBeenCalledWith(CZEN_BACKGROUND_CHECK_CC);
    });
  });
  it('query should redirect when Provider clicks on Next Button', async () => {
    await renderNoJobsRedirect(mocksList, CZEN_BACKGROUND_CHECK_CC);
    const nextBtn = screen.getByRole('button', { name: 'Next' });
    fireEvent.click(nextBtn);
    await waitFor(() => {
      expect(window.location.assign).toHaveBeenCalledWith(CZEN_BACKGROUND_CHECK_CC);
    });
  });
  it('query returns more than 1 jobSummaries results', async () => {
    await renderResultAndWaitFinished(mocksList);
    expect(screen.getByText('Companion Care Needed For My Mother In Round Rock')).toBeVisible();
  });
  it('query returns 1 jobSummary result', async () => {
    await renderResultAndWaitFinished(mocksList);
    expect(screen.getByText('Hands-on Care Needed For My Mother In Austin')).toBeVisible();
  });
  it('query returns Job posted with in 7 days', async () => {
    await renderResultAndWaitFinished(mocksList);
    expect(screen.getByText(/days ago/)).toBeVisible();
  });
  it('query returns Job posted with in one hour', async () => {
    await renderResultAndWaitFinished(mocksList);
    expect(screen.getByText(/hour ago/)).toBeVisible();
  });
});
