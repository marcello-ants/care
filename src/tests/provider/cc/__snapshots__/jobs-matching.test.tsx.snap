// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs matching page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-wrapper-1"
  >
    <div
      class="makeStyles-fade-2"
      style="position: relative; width: auto; margin-bottom: -24px; padding: 32px 40px 24px;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Next
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
